### PR TITLE
Add Error Boundary and Create Authorization Landing Page

### LINKED ISSUE #504: Unauthorized users are redirected to front page
**Describe the bug**
Unauthorized users are redirected to front page

**To Reproduce**
Not added as user to Flotilla and ISAR 
Log in to application 

**Expected behavior**
Should not be directed to frontpage, but receive message that you have no roles in this application

**Screenshots**
image

image

### LINKED ISSUE #505: Create landing page for non-authorized users
**Describe the bug**
Currently, all Equinor employees are authenticated when the log in to frontend. Unauthorized users should be redirected to a landing page where they are asked to apply for access.

**To Reproduce**
Log in to flotilla without being registered as a user in the relevant flotilla Enterprise application(service principal=
**Expected behavior**
Unauthorized users are expected to be redirected to a separate page, and they should not see anything if they naviage to other pages.

**Screenshots**

### PR DESCRIPTION
**Pull request by GodVenn**
Closes #504 
Closes #505

### CODE REVIEW

File: frontend/src/components/Pages/ErrorFallback.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+export const ErrorFallback = (error: Error) => {
+    switch (error.message) {
+        case '403 - ': {
+            return <UnauthorizedPage />
+        }
+        default: {
+            return (
+                <>
+                    Type = {error.name} - {error.message} - {error.cause}
```
Comment (GodVenn) [inline frontend/src/components/Pages/ErrorFallback.tsx:11]: This default message has too much information, suggesting reducing it
```suggestion
                    An unknown error has occured
```

**Review by GodVenn — COMMENTED** [review]
Looks nice!

File: frontend/src/components/Pages/HistoricMissionPage/HistoricMissionView.tsx [hunk 2]
```diff
@@ -38,18 +41,21 @@ export function HistoricMissionView({ refreshInterval }: RefreshProps) {
...
-                    .sort((a, b) =>
-                        compareDesc(
-                            a.endTime === undefined ? new Date(0) : new Date(a.endTime),
-                            b.endTime === undefined ? new Date(0) : new Date(b.endTime)
+        apiCaller
+            .getMissions()
+            .then((missions) => {
+                setCompletedMissions(
+                    missions
+                        .filter((m) => completedStatuses.includes(m.missionStatus))
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/HistoricMissionPage/HistoricMissionView.tsx:49]: Out of scope of this pr but I would refactor this to separate function

File: frontend/src/components/Pages/UnauthorizedPage/UnauthorizedPage.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    flex-direction: column;
+    padding-block: 20rem;
+    gap: 3rem;
+`
+
+const StyledText = styled.div`
+    font-family: Calibri;
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/UnauthorizedPage/UnauthorizedPage.tsx:14]: use Equinor font from EDS

File: frontend/src/components/Pages/ErrorFallback.tsx (hunk 1 above)
Comment (eivindsjovold) [reply frontend/src/components/Pages/ErrorFallback.tsx:11]: Is there something between exposing too much information and "An unknown error has occurred"?

Comment (GodVenn) [reply frontend/src/components/Pages/ErrorFallback.tsx:11]: I think exposing no information we can't control is the best for now, and we have to add more specific information pages for other error cases

File: frontend/src/components/Pages/ErrorFallback.tsx [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { UnauthorizedPage } from './UnauthorizedPage/UnauthorizedPage'
+import { UnknownErrorPage } from './UnknownErrorPage/UnknownErrorPage'
+
+export const ErrorFallback = (error: Error) => {
+    switch (error.message) {
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/ErrorFallback.tsx:5]: Does Error have a better property than message to switch on?

File: frontend/src/components/Pages/UnknownErrorPage/UnknownErrorPage.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { Typography } from '@equinor/eds-core-react'
+import styled from 'styled-components'
+
+const StyledUnauthorizedPage = styled.div`
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/UnknownErrorPage/UnknownErrorPage.tsx:4]:
```suggestion
const StyledUnknownErrorPage = styled.div`
```

**Review by eivindsjovold — COMMENTED** [review]
See comments

File: frontend/src/App.tsx [hunk 6]
```diff
@@ -1,10 +1,12 @@
 import { AuthenticatedTemplate, UnauthenticatedTemplate } from '@azure/msal-react'
 import { SignInPage } from './components/Pages/SignInPage/SignInPage'
 import { FlotillaSite } from 'components/Pages/FlotillaSite'
+import { ErrorBoundary } from 'react-error-boundary'
+import { ErrorFallback } from 'components/Pages/ErrorFallback'
 
 function App() {
     return (
-        <div>
+        <ErrorBoundary fallbackRender={({ error, resetErrorBoundary }) => ErrorFallback(error)}>
```
Comment (eivindsjovold) [inline frontend/src/App.tsx:9]: Type on error?

File: frontend/src/api/ApiCaller.tsx [hunk 7]
```diff
@@ -75,15 +75,17 @@ export class BackendAPICaller {
     async getRobots(): Promise<Robot[]> {
         const path: string = 'robots'
         const result = await this.GET<Robot[]>(path).catch((e) => {
-            throw new Error(`Failed to GET /${path}: ` + e)
+            console.error(`Failed to GET /${path}: ` + e)
```
Comment (eivindsjovold) [inline frontend/src/api/ApiCaller.tsx:78]: Type on e?

File: frontend/src/components/Pages/UnauthorizedPage/GoToAccessITButton.tsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import { Button } from '@equinor/eds-core-react'
+
+export const GoToAccessITButton = () => {
+    return (
+        <Button href="https://accessit.equinor.com" variant="contained">
+            Go to AccessIT
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/UnauthorizedPage/GoToAccessITButton.tsx:6]: Should this text be translatable? or not

File: frontend/src/components/Pages/UnauthorizedPage/GoToAccessITButton.tsx [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import { Button } from '@equinor/eds-core-react'
+
+export const GoToAccessITButton = () => {
+    return (
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/UnauthorizedPage/GoToAccessITButton.tsx:4]: Should this code be in Unauthorizedpage.tsx? Not sure

File: frontend/src/components/Pages/UnauthorizedPage/UnauthorizedPage.tsx [hunk 10]
```diff
@@ -0,0 +1,22 @@
+import { Typography } from '@equinor/eds-core-react'
+import styled from 'styled-components'
+import { GoToAccessITButton } from './GoToAccessITButton'
+
+const StyledUnauthorizedPage = styled.div`
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/UnauthorizedPage/UnauthorizedPage.tsx:5]: This styling is duplicated in two files, maybe refactor the page component to be reusable for both

**Review by eivindsjovold — CHANGES_REQUESTED** [review]

File: frontend/src/components/Pages/UnknownErrorPage/UnknownErrorPage.tsx [hunk 11]
```diff
@@ -1,20 +1,12 @@
...
+import { Text } from 'components/Contexts/LanguageContext'
 
 export const UnknownErrorPage = () => {
     var errorMessage = 'An unknown error has occurred'
     return (
-        <StyledUnknownErrorPage>
-            <Typography variant="h1">{errorMessage}</Typography>
-        </StyledUnknownErrorPage>
+        <StyledCenteredPage>
+            <Typography variant="h1">{Text(errorMessage)}</Typography>
```
Comment (eivindsjovold) [inline frontend/src/components/Pages/UnknownErrorPage/UnknownErrorPage.tsx:9]: Why do you construct Text?

Comment (eivindsjovold) [reply frontend/src/components/Pages/UnknownErrorPage/UnknownErrorPage.tsx:9]: I see, this is poor naming of the component from language context

**Review by eivindsjovold — APPROVED** [review]
LGTM :)

**Pull request merged**